src/core/map.js

```diff
@@ -373,6 +415,53 @@ geo.map = function (arg) {
 
   ////////////////////////////////////////////////////////////////////////////
   /**
+   * Get/set the map rotation.  The rotation is performed around the current
+   * view center.
+   *
+   * @param {Object} rotation angle in radians (positive is clockwise)
+   * @param {Object} origin is specified, rotate about this origin
+   * @param {boolean} ignoreRotationFunc if true, don't constrain the rotation.
+   * @returns {geo.map}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.rotation = function (rotation, origin, ignoreRotationFunc) {
+    if (rotation === undefined) {
+      return m_rotation;
+    }
+    rotation = fix_rotation(rotation, ignoreRotationFunc);
+    if (rotation === m_rotation) {
+      return m_this;
+    }
+    m_rotation = rotation;
+
+    var bounds = m_this.boundsFromZoomAndCenter(
+        m_zoom, m_center, m_rotation, null);
+    m_this.modified();
+
+    camera_bounds(bounds, m_rotation);
+
+    var evt = {
+      geo: {},
+      rotation: m_rotation,
+      screenPosition: origin ? origin.map : undefined
+    };
+
+    m_this.geoTrigger(geo.event.rotate, evt);
+
+    if (origin && origin.geo && origin.map) {
+      var shifted = m_this.gcsToDisplay(origin.geo);
+      m_this.pan({x: origin.map.x - shifted.x, y: origin.map.y - shifted.y});
+    } else {
+      m_this.pan({x: 0, y: 0});
+    }
+    /* Changing the rotation can change our minimum zoom */
```

Comment:
Say you have a window that is 512 x 512 and are clamping the zoom.  Without rotation, you can zoom out to level = 1.  You rotate 45 degrees.  Now, the width and height of the rotated map at level 1 is sqrt(2) * 512, so you need to be able to zoom out to level = 0.5 to see the entire map.